pymc/progress_bar.py

```diff
@@ -275,11 +412,16 @@ def __init__(
 
         progress_columns, progress_stats = step_method._progressbar_config(chains)
 
-        self._progress = self.create_progress_bar(
-            progress_columns,
-            progressbar=progressbar,
-            progressbar_theme=progressbar_theme,
-        )
+        if in_marimo_notebook():
```

Comment:
cache this result, imports are a tad slow

Reply:
python always has to check it again, but doesn't re-import no. But it's particularly bad when it's not installed:

```python
import numpy as np

def foo(x):
    return np.exp(x)

def bar(x):
    import numpy as np
    return np.exp(x)

def qux(x):
    try:
        import not_installed
    except ImportError:
        pass
    return np.exp(x)

%timeit foo(1)  # 592 ns ± 4.57 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
%timeit bar(1)  # 915 ns ± 3.59 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
%timeit qux(1)  # 57 μs ± 292 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

Reply:
Plus I don't know how much the `mo.running_in_marimo()` takes, but that's not my concern as a non-marimo user :)

Reply:
One observation on this front, [pep 810 has been accepted](https://peps.python.org/pep-0810/), so with the advent of lazy imports being a thing in the future this may be less of a concern.

Reply:
Yeah but that will take a while until our oldest compatible python version is covered